### PR TITLE
Increase magazine spawns

### PR DESCRIPTION
**Pull request by mugling**
Starts work on #15443

Extends `item_group` to support `magazine` and `ammo` fields specifying percentage chance to spawn reloadable items with a default magazine and/or default ammo. These fields have similar semantics to those provided in the new mapgen function from #15341. Deprecates the now redundant `guns_with_ammo` field.

Most monster drops that include gun drops now have a 100% chance of including a matching magazine and a 30% chance of matching ammo. A further PR will extend this to include optional randomisation of the quantity of ammo.

### CODE REVIEW

File: src/item_factory.cpp [hunk 1]
```diff
@@ -1424,7 +1424,6 @@ void Item_factory::load_item_group(JsonObject &jsobj, const Group_tag &group_id,
     Item_group *ig = dynamic_cast<Item_group *>(isd);
     if (subtype == "old") {
         ig = make_group_or_throw(isd, Item_group::G_DISTRIBUTION);
-        ig->with_ammo = jsobj.get_bool("guns_with_ammo", ig->with_ammo);
```
Comment (mugling) [inline src/item_factory.cpp]: Previously not applied to all `item_groups`

File: src/item_factory.cpp [hunk 2]
```diff
@@ -1433,6 +1432,9 @@ void Item_factory::load_item_group(JsonObject &jsobj, const Group_tag &group_id,
         jsobj.throw_error("unknown item group type", "subtype");
     }
 
+    ig->with_ammo = jsobj.get_int( "ammo", ig->with_ammo );
```
Comment (mugling) [inline src/item_factory.cpp]: `ammo` and `magazine` apply to both collections and distributions

**Comment by Rivet-the-Zombie** [comment]
Very nice!

**Comment by Coolthulhu** [comment]
The ammo spawns could be more random. Currently it's all or nothing. While it makes a lot of sense for someone to die after running out of ammo, those who die from other causes (infection) would probably have less than full magazine too.

**Comment by mugling** [comment]
Yes, this is an intended extension. I also want to eventually update mapgen so that it spawns an equivalent number of empty casings to match the rounds missing from the magazine

**Pull request merged**